### ISSUE TITLE
Extending Java Enum Customization to provide both Java Enum Name and Description

### ISSUE THREAD

**Issue by ddcruver** [joelittlejohn/jsonschema2pojo#1035]
I would like to extend the concept of javaEnumNames with a new extension named javaEnums to include the definition of the enum names and titles/descriptions to be inserted into JavaDocs as short and long descriptions.

This is an alternative to potential methods I have discussed on https://github.com/joelittlejohn/jsonschema2pojo/issues/1000.

Of the options I have discussed I believe this is the best because it is still compatible with JSON Schema (if the javaEnums field is ignored) and anything defined in javaEnums only effects generated classes.

So this:
```json
{
  "$id": "https://example.org/schemas/math/Digit.schema.json",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "type": "string",
  "javaType": "org.example.jsonschemas2pojo.math.Digit",
  "description": "All the digits.",
  "enum": [
    "One",
    "Two",
    "Three",
    "Four",
    "Five",
    "Six",
    "Seven",
    "Eight",
    "Nine"
  ],
  "javaEnumNames": ["I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX"],
  "required": [],
  "definitions": {},
  "additionalProperties": false
}
```

Would become this...
```json
{
  "$id": "https://example.org/schemas/math/Digit.schema.json",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "type": "string",
  "javaType": "org.example.jsonschemas2pojo.math.Digit",
  "description": "All the digits.",
  "enum": [
    "One",
    "Two",
    "Three",
    "Four",
    "Five",
    "Six",
    "Seven",
    "Eight",
    "Nine"
  ],
  "javaEnums": [
    {
      "name": "I",
      "description": "The first number."
    },
    {
      "name": "II",
      "description": "The second number."
    },
    {
      "name": "III",
      "description": "The third number."
    }
  ],
  "required": [],
  "definitions": {},
  "additionalProperties": false
}
```

The result would be something like this:
```java

package org.example.jsonschema2pojo.math;

import java.util.HashMap;
import java.util.Map;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum Digit {

	/**
	 * The first number.
	 */
	I("One"),

	/**
	 * The second number.
	 */
	II("Two"),

	/**
	 * The third number.
	 */
	III("Three"),

	/**
	 * The fourth number.
	 */
	IV("Four"),

	/**
	 * The fifth number.
	 */
	V("Five"),

	/**
	 * The sixth number.
	 */
	VI("Six"),

	/**
	 * The seventh number.
	 */
	VII("Seven"),

	/**
	 * The eighth number.
	 */
	VIII("Eight"),

	/**
	 * The ninth number.
	 */
	IX("Nine");
	private final String value;
	private final static Map<String, Digit> CONSTANTS = new HashMap<String, Digit>();

	static {
		for (Digit c: values()) {
			CONSTANTS.put(c.value, c);
		}
	}

	private Digit(String value) {
		this.value = value;
	}

	@Override
	public String toString() {
		return this.value;
	}

	@JsonValue
	public String value() {
		return this.value;
	}

	@JsonCreator
	public static Digit fromValue(String value) {
		Digit constant = CONSTANTS.get(value);
		if (constant == null) {
			throw new IllegalArgumentException(value);
		} else {
			return constant;
		}
	}

}

```

The title/description distinction should be used for a short and long description in the java docs as such, but either one could be optional.

```java
public enum Digit {

	/**
	 * The first number.
	 * <p>
	 * 1 (one, also called unit, unity, and (multiplicative) identity) is a number, and a numerical digit used to represent that number in numerals. It represents a single entity, the unit of counting or measurement. For example, a line segment of unit length is a line segment of length 1. It is also the first of the infinite sequence of natural numbers, followed by 2.
	 * </p>
	 */
	I("One"),
}
```

If inline comments are preferred then we could elect to only use the title if both are provided.

```java
public enum Digit {
	I("One"), // The first number
}
```

**Comment by joelittlejohn** [comment]
Implemented in #1041 and included in 1.0.2.

**Comment by joelittlejohn** [comment]
(Thanks!)
